### PR TITLE
Improve download function

### LINKED ISSUE #117: More robust download function
Some datasets are hundreds of gigabytes. Currently the default downloader provides no notion of progress, let alone support for resumption.

Surely there's a library out there that we can offload all of this to...

### PR DESCRIPTION
**Pull request by vdumoulin**
This PR adds a progress bar. Partially fixes #117.

### CODE REVIEW

**Comment by vdumoulin** [comment]
@dwf

File: fuel/downloaders/base.py [hunk 1]
```diff
@@ -11,6 +12,22 @@ class NeedURLPrefix(Exception):
     pass
 
 
+@contextmanager
+def progress_bar(name, maxval):
```
Comment (dwf) [inline fuel/downloaders/base.py]: nice use of context managers

**Comment by vdumoulin** [comment]
I'm debating adding an additional dependency (`httmock`) to mock HTTP requests. It allows to speed up the tests as well as craft responses that target specific cases in download helpers. Any thoughts?

**Comment by dwf** [comment]
Probably a good call (running tests offline is kind of important for working on the road, and the UCI people probably do not appreciate our Travis bot hitting their server frequently), but save that for another PR.

**Comment by bartvm** [comment]
For tests you can add whatever you want I think, because the user won't
have to install them. Just make sure to add them to req-travis-pip.txt
after https://github.com/mila-udem/fuel/pull/120 got merged

**Comment by bartvm** [comment]
Oh, and add them as an optional dependency in `setup.py`

**Comment by dwf** [comment]
IIRC there's some sort of "tests_require" thing in setup.py, no?

**Comment by bartvm** [comment]
https://github.com/mila-udem/blocks/blob/master/setup.py#L40-L44

**Pull request merged**